.travis.yml

```diff
@@ -0,0 +1,50 @@
+# .travis.yml
+
+language: python
+sudo: true
+
+env:
+  - LUA="lua=5.1"
+  - LUA="lua=5.2"
+  - LUA="lua=5.3"
+  - LUA="luajit=2.0"
+  - LUA="luajit=2.1"
+
+before_install:
+  - pip install hererocks
+  - hererocks lua_install -r^ --$LUA
+  - export PATH=$PATH:$PWD/lua_install/bin
+  - # --- installs development dependencies ---
+    # --- designed for debian linux systems ---
+  - set -e
+  - export CFLAGS="-Wno-error"
+  - # install cmake and git
+  - sudo apt-get install -y cmake git-core
+  - # install lua
+  - sudo apt-get install -y lua5.1 liblua5.1-0-dev luarocks
+  - # install json-c
+  - sudo apt-get install -y dh-autoreconf
+  - git clone https://github.com/json-c/json-c.git --depth=1 && cd json-c
+  - sh autogen.sh && ./configure && sudo make && sudo make install && cd ..
```

Comment:
I see a lot of duplication from this script:
https://github.com/openwisp/openwisp-config/blob/master/install-dev.sh

We should avoid this. Duplication is evil because it leads to maintenance issues.

Can you find a way to avoid it? For example, you could adapt the `install-dev.sh` script to accept an input parameter like `--ci` which if present will run commands that are specific for travis, that's just an example, try to make it in a way that the script can be used both by developers needing to install the dev env and for travis-ci.

Reply:
thanks, its working.